scipy/interpolate/_bsplines.py

```diff
@@ -445,34 +469,89 @@ def integrate(self, a, b, extrapolate=None):
         if extrapolate is None:
             extrapolate = self.extrapolate
 
-        if not extrapolate:
-            # shrink the integration interval, if needed
-            a = max(a, self.t[self.k])
-            b = min(b, self.t[-self.k - 1])
-
-            if self.c.ndim == 1:
-                # fast path: use FITPACK's routine (cf _fitpack_impl.splint)
-                t, c, k = self.tck
-                aint, wrk = _dierckx._splint(t, c, k, a, b)
-                return aint
-
         # prepare t & c
         self._ensure_c_contiguous()
 
+        # Swap integration bounds if needed
+        sign = 1
+        if b < a:
+            a, b = b, a
+            sign = -1
+
+        out = np.empty((2, prod(self.c.shape[1:])), dtype=self.c.dtype)
+        n = self.t.size - self.k - 1
+
         # compute the antiderivative
         c = self.c
         ct = len(self.t) - len(c)
         if ct > 0:
             c = np.r_[c, np.zeros((ct,) + c.shape[1:])]
         t, c, k = _fitpack_impl.splantider((self.t, c, self.k), 1)
 
-        # evaluate the diff of antiderivatives
-        x = np.asarray([a, b], dtype=np.float_)
-        out = np.empty((2, prod(c.shape[1:])), dtype=c.dtype)
-        _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
-                k, x, 0, extrapolate, out)
-        out = out[1] - out[0]
-        return out.reshape(c.shape[1:])
+        if extrapolate=='periodic':
+            # Split the integral into the part over period (can be several
+            # of them) and the remaining part.
+
+            ts, te = self.t[k], self.t[n]
+            period = te - ts
+            interval = b - a
+            n_periods, left = divmod(interval, period)
+
+            if n_periods > 0:
+                # evaluate the diff of antiderivatives
+                x = np.asarray([ts, te], dtype=np.float_)
+                _bspl.evaluate_spline(t, c.reshape(c.shape[0], -1),
+                        k, x, 0, False, out)
+                new = out[1] - out[0]
+                new *= n_periods
+            else:
+                new.fill(0)
+
+            # Map a to [ts, te], b is always a + left.
+            a = ts + (a - ts) % period
+            b = a + left
+
+            # If b <= te then we need to integrate over [a, b], otherwise
+            # over [a, te] and from xs to what is remained.
+            remainder_int = np.empty_like(out)
```

Comment:
Another remainder to use `out` and not to allocate more memory. Are there some flaws to reuse it which I don't see?